### PR TITLE
add mass action jump support

### PR DESCRIPTION
**Pull request by isaacsas**
~Still needs tests for zero order, second order and decay reactions.~ Should be set now.

### CODE REVIEW

**Comment by isaacsas** [comment]
Also needs a new release of DiffEqJump with the iszero fix.

File: src/systems/jumps/jumpsystem.jl [hunk 1]
```diff
@@ -42,6 +42,29 @@ function assemble_crj(js, crj, statetoid)
     ConstantRateJump(rate, affect)
 end
 
+function assemble_maj(js, maj::MassActionJump{U,Vector{Pair{V,W}},Vector{Pair{V2,W2}}}, 
+                      statetoid, ptoid, p, pcontext) where {U,V,W,V2,W2}
+    pval = Base.eval(pcontext, Expr(maj.scaled_rates))
```
Comment (ChrisRackauckas) [inline src/systems/jumps/jumpsystem.jl:47]: I assume this is the evaluation?

Comment (isaacsas) [reply src/systems/jumps/jumpsystem.jl:47]: Yeah

Comment (isaacsas) [reply src/systems/jumps/jumpsystem.jl:47]: If you have suggestions on a better way to do this please do let me know...

Comment (ChrisRackauckas) [reply src/systems/jumps/jumpsystem.jl:47]: We can go with it for now while we work on a better solution.

**Comment by isaacsas** [comment]
@ChrisRackauckas Assuming tests pass I think this is good to go.
